cpp/src/barretenberg/dsl/types.hpp

```diff
@@ -0,0 +1,63 @@
+#pragma once
+#include "barretenberg/plonk/composer/ultra_composer.hpp"
+
+#include "barretenberg/plonk/proof_system/prover/prover.hpp"
+#include "barretenberg/stdlib/primitives/bigfield/bigfield.hpp"
+#include "barretenberg/stdlib/primitives/biggroup/biggroup.hpp"
+#include "barretenberg/stdlib/primitives/bit_array/bit_array.hpp"
+#include "barretenberg/stdlib/primitives/bool/bool.hpp"
+#include "barretenberg/stdlib/primitives/packed_byte_array/packed_byte_array.hpp"
+#include "barretenberg/stdlib/primitives/byte_array/byte_array.hpp"
+#include "barretenberg/stdlib/primitives/uint/uint.hpp"
+#include "barretenberg/stdlib/primitives/witness/witness.hpp"
+#include "barretenberg/stdlib/primitives/bigfield/bigfield.hpp"
+#include "barretenberg/stdlib/primitives/biggroup/biggroup.hpp"
+#include "barretenberg/stdlib/commitment/pedersen/pedersen.hpp"
+#include "barretenberg/stdlib/commitment/pedersen/pedersen_plookup.hpp"
+#include "barretenberg/stdlib/merkle_tree/hash_path.hpp"
+#include "barretenberg/stdlib/encryption/schnorr/schnorr.hpp"
+#include "barretenberg/stdlib/primitives/curves/bn254.hpp"
+#include "barretenberg/stdlib/primitives/curves/secp256k1.hpp"
+#include "barretenberg/stdlib/primitives/memory/rom_table.hpp"
+namespace acir_format {
+
+using Composer = plonk::UltraComposer;
+
+using Prover = std::conditional_t<
```

Comment:
Thinking on adding recursion later, how simple would it be to substitute `UltraWithKeccakProver` with `UltraProver` to generate a proof using pedersen-blake3, while still maintaining an `UltraWithKeccakProver` in order to be able to verify uses the solidity verifier? Would we need two different `Prover` types?

Reply:
Yes I think we should declare two provers:
`IntermediateProver = UltraProver`
`FinalProver = UltraWithKeccakProver`
Better to use explicit prover type when testing too.

Reply:
Agree